### DISCUSSION TITLE
NextCloud / DoveCot

### DISCUSSION THREAD

**Discussion by peter4200** [Z-Hub/Z-Push#49] [General]
Hi there
Hoping someone can give me a hint or instruction on how to configure z-push (2.7.1) on Ubuntu 22.04 to work with dovecot (2.3.16) and nextcloud (27.1.5). I need some settings or instructions on how to setup a CombinedBackend to support IMAP, CalDAV and CardDAV

Thanks
Peter

**Comment by matidau** [comment]
Hi @peter4200

If you are building an Imap/caldav/carddav server, I would recommend that you install with MiaB https://github.com/mail-in-a-box/mailinabox as this has Z-Push configured out of the box and is great as a starting point.

If you want to do your own Imap set up, review the zpush.sh from the MiaB project as this has most of the configuration needed. The only other part is the script to setup nginx.

Cheers, 
Mat
